### ISSUE TITLE
QuartzNet O1 divergence

### ISSUE THREAD

**Issue by fminkin** [NVIDIA/NeMo#272]
Hello! 

I'm trying to launch QuartzNet on dataset, where jasper does just fine, we've also checked wavs / features correctness. After several epochs training in mixed precision diverges, cause nans are appearing in logits. 
Batch size is 64 x 8 (GPU)s x 10 (virtual batch size).
We've also tried decreasing learning rate almost by 10 times, training diverges.

I've also seen OpenSeq2Seq implementation and it has "dtype": "float32" in it. I'm wondering, is there something you can share about this behaviour, it seems you've faced something similar in openseq2seq framework?

**Comment by borisgin** [comment]
Do you have NaN if you train w/o virtual batch size?

**Comment by fminkin** [comment]
We launched experiment w/o virtual batch size, it ends up having lots of nans, network won't even train.


Here is a tensorboard plot.

<img width="1007" alt="Screenshot 2020-01-20 at 11 47 25" src="https://user-images.githubusercontent.com/8925958/72712174-8ebd2c80-3b7b-11ea-8cc4-6afade53c22e.png">

**Comment by fminkin** [comment]
Yes, we do, here is tensoboard for loss and console output

`2020-01-20 10:55:42,554 - INFO - ==========>>>>>>Evaluation Loss dataset: 69.41630554199219
2020-01-20 10:55:42,554 - INFO - ==========>>>>>>Evaluation WER dataset:  103.27%
2020-01-20 10:55:42,555 - INFO - Evaluation time: 5.935473203659058 seconds
2020-01-20 10:56:56,094 - INFO - Step: 2100
2020-01-20 10:56:56,099 - INFO - Loss: 100.46298217773438
2020-01-20 10:56:56,099 - INFO - training_batch_WER:  100.00%
2020-01-20 10:56:56,099 - INFO - Step time: 0.5234289169311523 seconds
2020-01-20 10:58:09,265 - INFO - Step: 2200
2020-01-20 10:58:09,271 - INFO - Loss: 148.91075134277344
2020-01-20 10:58:09,272 - INFO - training_batch_WER:  99.26%
2020-01-20 10:58:09,272 - INFO - Step time: 0.8416743278503418 seconds
2020-01-20 10:59:22,426 - INFO - Step: 2300
2020-01-20 10:59:22,431 - INFO - Loss: 110.98063659667969
2020-01-20 10:59:22,431 - INFO - training_batch_WER:  100.00%
2020-01-20 10:59:22,431 - INFO - Step time: 0.5927708148956299 seconds
2020-01-20 11:00:16,715 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:21,478 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:21,694 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:23,556 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:23,783 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:24,557 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:25,313 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:26,789 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:26,988 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:27,209 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:27,468 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:28,826 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:29,030 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:30,714 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:30,915 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:31,169 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:33,716 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:34,421 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:34,644 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:34,830 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:35,019 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:35,759 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:39,502 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:39,691 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:39,892 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:40,094 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:40,306 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:42,553 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:43,556 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:43,747 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:44,642 - INFO - Step: 2400
2020-01-20 11:00:44,646 - INFO - Loss: 74.99761199951172
2020-01-20 11:00:44,646 - INFO - training_batch_WER:  100.00%
2020-01-20 11:00:44,646 - INFO - Step time: 0.891425371170044 seconds
2020-01-20 11:00:44,838 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:45,036 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:45,264 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:48,555 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:48,740 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:48,954 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:51,340 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:54,558 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:54,759 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:54,975 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:55,160 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:55,394 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:55,594 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:55,786 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:55,973 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:56,167 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:56,405 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:00:57,789 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:01,427 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:04,716 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:11,528 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:11,785 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:12,036 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:15,683 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:16,796 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:17,475 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:17,665 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:19,404 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:19,662 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:24,024 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:24,217 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)
2020-01-20 11:01:24,883 - WARNING - WARNING: Loss is NaN or inf tensor(nan, device=‘cuda:0’, grad_fn=<MeanBackward0>)`

<img width="428" alt="Screenshot 2020-01-20 at 17 58 59" src="https://user-images.githubusercontent.com/8925958/72742548-0494b880-3bbb-11ea-90b9-3a69e98f3d35.png">

**Comment by borisgin** [comment]
Can you attach complete log with all training parameters ( optimizer, LR, weight decay etc) please?

**Comment by fminkin** [comment]
Sorry for the delay, here's folder with files:
https://drive.google.com/drive/folders/1NdKa7kz5IUbY9Q4rtOqeBtBEjU76INR_?usp=sharing

and lr, wd, optimizer are here:
`    parser.set_defaults(
        checkpoint_dir=None,
        optimizer="novograd",
        batch_size=32,
        eval_batch_size=64,
        lr=0.02,
        weight_decay=0.001,
        amp_opt_level="O1",
        eval_freq=1000,
        train_eval_freq=100,
        checkpoint_save_freq=500,
        iter_per_step=1,
        stop_on_nan_loss=False,
        create_tb_writer=True
    )`
